pkg/kubectl/resource_filter.go

```diff
@@ -0,0 +1,110 @@
+package kubectl
+
+import (
+	"fmt"
+	"io"
+	"reflect"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+type FilterFunc func(runtime.Object) (bool, error)
+
+type FilterOptions struct {
+	filterMap    map[reflect.Type]reflect.Value
```

Comment:
Why are you keeping this a `reflect.Value` don't you know that it is a `FilterFunc` here?